### PR TITLE
[re-architectured generator]: Using ecr instead of liquid

### PR DESCRIPTION
**Pull request by meatball133**

### CODE REVIEW

File: bin/generator.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# error prints an error to STDERR and exits.
+
+if [ -z "$1" ] ; then
+    echo "usage: ./bin/template.sh exercise-slug "
+    exit 1
+fi
+
+slug="$1"
+
+cat "./exercises/practice/${slug}/.meta/test_template.ecr" > "./bin/templates/template.ecr"
```
Comment (ErikSchierboom) [inline bin/generator.sh:14]: Shouldn't this be the other way around?

```suggestion
cat "./bin/templates/template.ecr" > "./exercises/practice/${slug}/.meta/test_template.ecr"
```

File: bin/generator_help.cr [hunk 2]
```diff
@@ -100,4 +89,37 @@ class GeneratorHelp
...
+
+  def to_snake(input)
+    result = ""
+    input.each_char do |x|
+      result += x.upcase == x ? "_#{x.downcase}" : x
+    end
+    result
+  end
+
+  def to_captlized(input)
```
Comment (ErikSchierboom) [inline bin/generator_help.cr:101]:
```suggestion
  def to_capitalized(input)
```

**Review by ErikSchierboom — APPROVED** [review]

File: bin/generator.sh (hunk 1 above)
Comment (meatball133) [reply bin/generator.sh:14]: No it should be that way, the script takes the template for each exercise and move it to a set folder.

File: bin/generator.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+if [ -z "$1" ] ; then
+    echo "usage: ./bin/template.sh exercise-slug "
+    exit 1
+fi
+
+slug="$1"
+
+if test ! -d example_directory; then
+  mkdir example_directory
+fi
```
Comment (ErikSchierboom) [inline bin/generator.sh:16]: Did you want to create the `bin/templates` directory here?

**Pull request merged**